### PR TITLE
antool: add fault injection tests

### PR DESCRIPTION
**Pull request by ldorau**
It will increase antool's code coverage to more than 98%.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/347)

### CODE REVIEW

File: tests/antool/test-fi.sh [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+check
+
+# test succeeded
+
+# copy coverage report
+[ "$COVERAGE" == "1" -a -f $COVERAGE_REPORT ] && cp -f $COVERAGE_REPORT ..
+
+# remove the temporary test directory
+cd ..
+rm -rf $DIR_NAME
```
Comment (marcinslusarz) [inline tests/antool/test-fi.sh]: Please remove files without using "-rf" flag. It's very dangerous. Also please validate that all variables used in rm are not empty.

Comment (ldorau) [reply tests/antool/test-fi.sh]: Done

**Comment by ldorau** [comment]
IMPORTANT: it will have to be rebased after the PRs are merged:
- https://github.com/pmem/pmemfile/pull/340
- https://github.com/pmem/pmemfile/pull/341
- https://github.com/pmem/pmemfile/pull/346,

Please do NOT merge it earlier.

**Comment by marcinslusarz** [comment]
*[tests/antool/test-fi.sh, line 290 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/347#-KtpESU58VN3-PxWlsbk-r1-290:-Ktv2N_wguPQUNWBM8hQ:b3u6j5q) ([raw file](https://github.com/pmem/pmemfile/blob/f803c937b33cbc2f8deb0696641112db69587872/tests/antool/test-fi.sh#L290)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Done
</blockquote></details>

So now you are using -rf just with 2 commands. Not really an improvement.

I was thinking about something like that:
```
if [ -z "${DIR_NAME}" ]; then
  exit 1
fi
rm ${DIR_NAME}/*
rmdir ${DIR_NAME}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by ldorau** [comment]
Review status: 0 of 59 files reviewed at latest revision, 1 unresolved discussion.

---

*[tests/antool/test-fi.sh, line 290 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/347#-KtpESU58VN3-PxWlsbk-r1-290:-Ktv63L-wL7ZB3nrkRGX:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/f803c937b33cbc2f8deb0696641112db69587872/tests/antool/test-fi.sh#L290)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

So now you are using -rf just with 2 commands. Not really an improvement.

I was thinking about something like that:
```
if [ -z "${DIR_NAME}" ]; then
  exit 1
fi
rm ${DIR_NAME}/*
rmdir ${DIR_NAME}
```


</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by llugin** [comment]
Review status: 0 of 59 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/tools/antool/syscall.py, line 237 at r3](https://reviewable.io:443/reviews/pmem/pmemfile/347#-KtvJ9rFavqcHQq0U1Vz:-KtvJ9rFavqcHQq0U1W-:b-fzhsez) ([raw file](https://github.com/pmem/pmemfile/blob/d8007b84b201e7521de80b0550721d0df914163d/src/tools/antool/syscall.py#L237)):*
> ```Python
>             self.log_print("corrupted entry packet information of syscall {0:s}:".format(self.name), debug)
>             self.log_print("0x{0:016X} 0x{1:016X} {2:s} {3:s} [corrupted entry packet]"
>                            .format(self.time_start, self.pid_tid, self.__str, self.name), debug)
> ```

I found '__str' name confusing, it's very similiar to built-in method __str__, while it's just a stripe :P. Maybe rename to __stripe?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by llugin** [comment]
Review status: 0 of 59 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/tools/antool/syscall.py, line 237 at r3](https://reviewable.io:443/reviews/pmem/pmemfile/347#-KtvJ9rFavqcHQq0U1Vz:-KtvK2htrZtIJhUwqfws:bn0pain) ([raw file](https://github.com/pmem/pmemfile/blob/d8007b84b201e7521de80b0550721d0df914163d/src/tools/antool/syscall.py#L237)):*
<details><summary><i>Previously, llugin wrote…</i></summary><blockquote>

I found '__str' name confusing, it's very similiar to built-in method __str__, while it's just a stripe :P. Maybe rename to __stripe?
</blockquote></details>

by __str__ I meant  __ str __  (apparently reviewable formats double underscores to bold)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by ldorau** [comment]
Review status: 0 of 59 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/tools/antool/syscall.py, line 237 at r3](https://reviewable.io:443/reviews/pmem/pmemfile/347#-KtvJ9rFavqcHQq0U1Vz:-KtvcaQXjFqPfzmUEUaV:b-896fix) ([raw file](https://github.com/pmem/pmemfile/blob/d8007b84b201e7521de80b0550721d0df914163d/src/tools/antool/syscall.py#L237)):*
<details><summary><i>Previously, llugin wrote…</i></summary><blockquote>

by __str__ I meant  __ str __  (apparently reviewable formats double underscores to bold)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by sarahjelinek** [comment]
Reviewed 4 of 187 files at r1, 1 of 9 files at r2, 1 of 2 files at r4.
Review status: 6 of 59 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[tests/antool/output-err-fi-1.log.match, line 0 at r4](https://reviewable.io:443/reviews/pmem/pmemfile/347#-Ktx37T_L-eAdX4k9cHV:-Ktx37T_L-eAdX4k9cHW:b33ef6z) ([raw file](https://github.com/pmem/pmemfile/blob/c4073986f575c0cee81e3980d1c574287342972a/tests/antool/output-err-fi-1.log.match#L0)):*
Empty file?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by ldorau** [comment]
Review status: 6 of 59 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[tests/antool/output-err-fi-1.log.match, line  at r4](https://reviewable.io:443/reviews/pmem/pmemfile/347#-Ktx37T_L-eAdX4k9cHV:-Ku3CZRrLYP6jrfEtOEM:buwy8oa) ([raw file](https://github.com/pmem/pmemfile/blob/c4073986f575c0cee81e3980d1c574287342972a/tests/antool/output-err-fi-1.log.match#L)):*
<details><summary><i>Previously, sarahjelinek (Sarah Jelinek) wrote…</i></summary><blockquote>

Empty file?
</blockquote></details>

Yes, the same like here:
https://reviewable.io/reviews/pmem/pmemfile/341#-KtmoipPBqoMyQbWufpK

An empty match file means we check that there is no stderr (in this case) output for the test.

---

*[tests/antool/test-fi.sh, line 290 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/347#-KtpESU58VN3-PxWlsbk-r1-290:-Ku3BUA6mUVHZYHC5a4h:bb726fr) ([raw file](https://github.com/pmem/pmemfile/blob/f803c937b33cbc2f8deb0696641112db69587872/tests/antool/test-fi.sh#L290)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Ping

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by marcinslusarz** [comment]
*[tests/antool/test-fi.sh, line 290 at r1](https://reviewable.io:443/reviews/pmem/pmemfile/347#-KtpESU58VN3-PxWlsbk-r1-290:-KuKV3slvnLEwQ4dY4PY:bki8i2x) ([raw file](https://github.com/pmem/pmemfile/blob/f803c937b33cbc2f8deb0696641112db69587872/tests/antool/test-fi.sh#L290)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Ping
</blockquote></details>

I still don't like it, but whatever.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347)*

**Comment by sarahjelinek** [comment]
:lgtm:

---

Review status: 4 of 77 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/347#-:-KuM21ZKrvyryEWYsGLr:bnfp4nl)*

**Pull request merged**